pkg/kn/commands/revision/delete.go

```diff
@@ -67,7 +103,27 @@ func NewRevisionDeleteCommand(p *commands.KnParams) *cobra.Command {
 			return nil
 		},
 	}
+	flags := RevisionDeleteCommand.Flags()
+	flags.StringVarP(&pruneFilter, "prune", "p", "", "Remove unreferenced revisions for a given service in a namespace.")
+	flags.Bool("prune-all", false, "Remove all unreferenced revisions in a namespace.")
```

Comment:
I'd suggest to add new variable `var pruneAll bool` variable to store the value to make the code more uniform. Then get rid of L49.
```
pruneAll, err := cmd.Flags().GetBool("prune-all")
```